wherehows-web/app/components/avatars/avatar-image.ts

```diff
@@ -37,16 +39,15 @@ export default class AvatarImage extends Component {
   @attribute
   onerror = (): void => {
     // Change avatar if an error occurs when the image is loaded
-    this.onImageFallback();
+    run(() => get(this, 'onImageFallback').perform());
   };
 
   /**
-   * Sets the fallback image for an avatar
+   * Task to set the fallback image for an avatar
+   * @type Task<void, (a?: {}) => TaskInstance<void>>
    * @memberof AvatarImage
    */
-  onImageFallback(): void {
-    if (!this.isDestroyed) {
-      set(this, 'src', this.avatar.imageUrlFallback);
-    }
-  }
+  onImageFallback = task(function*(this: AvatarImage): IterableIterator<void> {
```

Comment:
I would like to understand a little bit more how this works and why it is a task?